### PR TITLE
Adding the github_team_membership resource for #2

### PR DESCRIPTION
**Pull request by JacobASeverson**
This commit adds a resource that controls membership for
teams within organizations.
- Implemented the resource
- Added acceptance testing
- Added a page to the Terraform website for the resource

### CODE REVIEW

File: website/source/docs/providers/github/r/team_membership.html.markdown [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `team_id` - (Required) The Github team id
+* `username` - (Required) The user to add to the organization.
+* `role` - (Required) The role of the user within the team.
```
Comment (johnrengelman) [inline website/source/docs/providers/github/r/team_membership.html.markdown]: I think we should default this to member. Perhaps do the same for organization membership. Default them to the lowest privilege.

Comment (JacobASeverson) [inline website/source/docs/providers/github/r/team_membership.html.markdown]: Yeah that's a good point

**Comment by JacobASeverson** [comment]
turns out team membership was optional and defaulted, I just documented it on the website page wrong

**Comment by JacobASeverson** [comment]
Closing - these commits are now part of #10

**Pull request closed without merging**